src/main.zig

```diff
@@ -71,15 +71,17 @@ pub fn main() !void {
         , .{input}));
 
         // We request completions from zls
-        const completions: []const zls.types.CompletionItem = (try server.completionHandler(.{
+
+        const emptyList: zls.Server.ResultType("textDocument/completion") = .{ .CompletionList = zls.types.CompletionList{ .isIncomplete = true, .items = &.{} } };
+        const completions: []const zls.types.CompletionItem = (try server.completionHandler(allocator, .{
             .textDocument = .{
                 .uri = "file:///C:/Programming/Zig/zls-as-lib-demo/src/bs.zig",
             },
             .position = .{
                 .line = 4,
-                .character = @intCast(u32, 4 + input.len),
+                .character = @intCast(4 + input.len),
             },
-        }) orelse zls.types.CompletionList{ .isIncomplete = true, .items = &.{} }).items;
+        }) orelse emptyList).?.CompletionList.items;
```

Comment:
the unwrap is needed here because `zls.Server.ResultType("textDocument/completion")` is the following type:
https://github.com/zigtools/zls/blob/5bfff2a4b9ee01a7bab5fc26fad6e174f289c28d/src/lsp.zig#L7441-L7445
```zig
?union(enum) {
    array_of_CompletionItem: []const CompletionItem,
    CompletionList: CompletionList,
    pub usingnamespace UnionParser(@This());
}
```
This means that `((try server.completionHandler(...)) orelse emptyList)` resolves to an optional because `emptyList` is optional.